httprouter/apirest/apirest.go

```diff
@@ -57,6 +57,33 @@ type APIhandler = func(*APIdata, *httprouter.HTTPContext) error
 // ErrorMsg is the error returned by bearer std API
 type ErrorMsg struct {
 	Error string `json:"error"`
+	Code  int    `json:"code"`
+}
+
+// APIerror is used by handler functions to wrap errors, assigning a unique error code
+// and also specifying which HTTP Status should be used.
+type APIerror struct {
+	Message    string
+	Code       int
+	HTTPstatus int
+}
+
+// Error returns the Message contained inside the APIerror
+func (e APIerror) Error() string {
+	return e.Message
+}
+
+// JSON returns a serialized ErrorMsg using APIerror.Error() and APIerror.Code
+func (e APIerror) JSON() []byte {
+	data, err := json.Marshal(&ErrorMsg{
```

Comment:
No need to use a special struct `ErrorMsg` you can use APIerror if you declare it this way:

```go
type APIerror struct {
	Message    string `json:"error"`
	Code       int `json:"code"`
	HTTPstatus int `json:"-"`
}
```